lib/parsing/Cargo.toml

```diff
@@ -0,0 +1,21 @@
+[package]
+name = "parsing"
```

Comment:
I'm a bit confused by this crate. It doesn't seem to contain any parsing code. 

Is this moving the `Value` type  out of `vector-core` or creating a new `Value` type?

Reply:
Ah, right. This package will contain some shared parsing logic(ruby hash, xml etc). For example, if there is a vrl function which does the same thing as one of the grok filters we put the shared logic there.

Reply:
Agreed, I think "parsing" feels like too generic a name (especially hosted directly under `lib`).

I'd probably keep grok and VRL code separate, even if there's some redundancy/overlap there, unless there's a strong reason for this to be coupled.